src/renderer/features/accounts-structure/components/CustomEdge.tsx

```diff
@@ -34,6 +35,23 @@ export const CustomEdge = ({
   const pathType = useUnit(accountsStructureModel.$pathType);
   const edgeType = useUnit(accountsStructureModel.$edgeType);
   const highlightedNodesIds = useUnit(accountsStructureModel.$highlightedNodesIds);
+  const viewport = useUnit(accountsStructureModel.$viewport);
+
+  const isOutsideViewport = useMemo(() => {
+    // Transform source and target coordinates to screen coordinates
+    const sourceScreenX = sourceX * viewport.zoom + viewport.x;
+    const sourceScreenY = sourceY * viewport.zoom + viewport.y;
+    const targetScreenX = targetX * viewport.zoom + viewport.x;
+    const targetScreenY = targetY * viewport.zoom + viewport.y;
+
+    // Check if both source and target are outside viewport (assuming 0,0 to window dimensions)
+    return (
+      (sourceScreenX < 0 && targetScreenX < 0) ||
+      (sourceScreenY < 0 && targetScreenY < 0) ||
+      (sourceScreenX > window.innerWidth && targetScreenX > window.innerWidth) ||
```

Comment:
rect-flow is operating in context of canvas content, not window. Can you access container element? Or some other viewport or dimensions data?